torch/csrc/jit/script/init.cpp

```diff
@@ -266,6 +296,101 @@ FunctionSchema getSchemaWithNameAndDefaults(
       schema.is_varret());
 }
 
+static Decl computeOverloadDecl(
+    const Decl& overload_decl,
+    const Decl& impl_decl,
+    const FunctionDefaults& defaults) {
+  std::vector<Param> adjusted_params;
+  const auto& new_params = overload_decl.params();
+  const auto& old_params = impl_decl.params();
+
+  // following PEP specification that the following should work:
+  // @overload
+  // def mouse_event(x1: int, y1: int) -> ClickEvent: ...
+  // ...
+  // def mouse_event(x1: int, y1: int, x2: Optional[int] = None, y2:
+  // Optional[int] = None)
+  TORCH_CHECK(
+      new_params.size() <= old_params.size(),
+      "Overload should not have more parameters than implementation function",
+      overload_decl.range(),
+      impl_decl.range());
+
+  for (size_t i = 0; i < new_params.size(); ++i) {
+    auto overload_name = new_params[i].ident().name();
+    auto impl_name = old_params[i].ident().name();
+    if (overload_name != impl_name) {
+      throw ErrorReport(overload_decl.range())
+          << "Overload parameters must have the same names. "
+          << "Found " << overload_name << " and " << impl_name
+          << " on argument " << i;
+    }
+    adjusted_params.push_back(new_params[i]);
+  }
+  for (size_t i = new_params.size(); i < old_params.size(); ++i) {
+    if (!defaults.count(old_params[i].ident().name())) {
+      throw ErrorReport(impl_decl.range())
+          << "Expected to find default parameter on argument"
+          << old_params[i].ident().name()
+          << " because it is not defined on the overloaded declaration";
+    }
+    if (!old_params[i].type().present()) {
+      throw ErrorReport(impl_decl.range())
+          << "Parameters not specified on the overloaded declaration must have a type annotation in the implementation function."
+          << " Did not find type for param " << old_params[i].ident().name();
+    }
+    adjusted_params.push_back(old_params[i]);
+  }
+  return Decl::create(
+      overload_decl.range(),
+      List<Param>::create(overload_decl.range(), adjusted_params),
+      overload_decl.return_type());
+}
+
+void checkSchema(const FunctionSchema& schema, const SourceRange& range) {
```

Comment:
I tried this in Function Schema originally, but i failed on a bunch of functions for reasons I couldn't really understand. I can debug this more...

Reply:
It fails because symbolic_script registers an invalid schema:

```def __interpolate_0(input,
                            size: Optional[int],
                            scale_factor: Optional[List[float]],
                            mode: str='nearest',
                            align_corners: Optional[bool]):```

Reply:
ah, can we fold that fix into here? Seems bad to have it out there lol